orderbook/src/api.rs

```diff
@@ -41,66 +36,62 @@ pub fn handle_all_routes(
     let get_order = get_order_by_uid::get_order_by_uid(orderbook.clone());
     let get_solvable_orders = get_solvable_orders::get_solvable_orders(orderbook.clone());
     let get_trades = get_trades::get_trades(database);
-    let cancel_order = cancel_order::cancel_order(orderbook.clone());
+    let cancel_order = cancel_order::cancel_order(orderbook);
     let get_amount_estimate = get_markets::get_amount_estimate(price_estimator.clone());
     let get_fee_and_quote_sell =
         get_fee_and_quote::get_fee_and_quote_sell(fee_calculator.clone(), price_estimator.clone());
     let get_fee_and_quote_buy =
         get_fee_and_quote::get_fee_and_quote_buy(fee_calculator, price_estimator.clone());
-    let get_user_orders = get_user_orders::get_user_orders(orderbook);
     let cors = warp::cors()
         .allow_any_origin()
         .allow_methods(vec!["GET", "POST", "DELETE", "OPTIONS", "PUT", "PATCH"])
         .allow_headers(vec!["Origin", "Content-Type", "X-Auth-Token", "X-AppId"]);
     let routes_with_labels = warp::path!("api" / "v1" / ..).and(
-        (create_order.map(|reply| LabelledReply::new(reply, "create_order")))
-            .or(get_orders.map(|reply| LabelledReply::new(reply, "get_orders")))
-            .unify()
-            .or(fee_info.map(|reply| LabelledReply::new(reply, "fee_info")))
-            .unify()
-            .or(legacy_fee_info.map(|reply| LabelledReply::new(reply, "legacy_fee_info")))
-            .unify()
-            .or(get_order.map(|reply| LabelledReply::new(reply, "get_order")))
-            .unify()
-            .or(get_solvable_orders.map(|reply| LabelledReply::new(reply, "get_solvable_orders")))
-            .unify()
-            .or(get_trades.map(|reply| LabelledReply::new(reply, "get_trades")))
-            .unify()
-            .or(cancel_order.map(|reply| LabelledReply::new(reply, "cancel_order")))
-            .unify()
-            .or(get_amount_estimate.map(|reply| LabelledReply::new(reply, "get_amount_estimate")))
-            .unify()
-            .or(get_fee_and_quote_sell
-                .map(|reply| LabelledReply::new(reply, "get_fee_and_quote_sell")))
-            .unify()
-            .or(get_fee_and_quote_buy
-                .map(|reply| LabelledReply::new(reply, "get_fee_and_quote_buy")))
-            .unify()
-            .or(get_user_orders.map(|reply| LabelledReply::new(reply, "get_user_orders")))
-            .unify(),
+        (create_order.with(handle_metrics("create_order")))
+            .or(get_orders.with(handle_metrics("get_orders")))
+            .or(fee_info.with(handle_metrics("fee_info")))
```

Comment:
This looks much nicer, could you elaborate on how the api metrics change is related to this. Or even why we no longer need "unify"?